File: ios/sources/commands/ARESQuadraticCurveToCommand.m
//
//  ARESQuadraticCurveToCommand.m
//  ares
//
//  Created by PonyCui on 2018/11/29.
//  Copyright © 2018年 XT Studio. All rights reserved.
//

#import "ARESQuadraticCurveToCommand.h"

@implementation ARESQuadraticCurveToCommand

- (instancetype)initWithCpx:(float)cpx cpy:(float)cpy x:(float)x y:(float)y
{
    self = [super init];
    if (self) {
        _cpx = cpx;
        _cpy = cpy;
        _x = x;
        _y = y;
        [[ARESBeginPathCommand currentPath] addQuadCurveToPoint:CGPointMake(x, y)
                                                   controlPoint:CGPointMake(cpx, cpy)];
    }
    return self;
}

- (void)draw:(CGContextRef)ctx {
    CGContextAddQuadCurveToPoint(ctx, self.cpx, self.cpy, self.x, self.y);
}

@end

File: android/ares/src/main/java/com/xt/ares/commands/ARESQuadraticCurveToCommand.kt
package com.xt.ares.commands

import android.graphics.Canvas
import com.xt.ares.ARESCommand
import com.xt.ares.ARESView

class ARESQuadraticCurveToCommand(val cpx: Double, val cpy: Double, val x: Double, val y: Double): ARESCommand() {

    init {
        ARESBeginPathCommand.sharedPath.quadTo(cpx.toFloat(), cpy.toFloat(), x.toFloat(), y.toFloat())
    }

    override fun draw(view: ARESView, canvas: Canvas) {
        super.draw(view, canvas)
        view.currentPath.quadTo(cpx.toFloat(), cpy.toFloat(), x.toFloat(), y.toFloat())
    }

}